homeassistant/components/group/__init__.py

```diff
@@ -309,8 +309,7 @@ async def reload_service_handler(service: ServiceCall) -> None:
         tasks = [
             entity.async_remove()
             for entity in component.entities
-            if entity.entity_id.startswith("group.")
-            and not cast(Group, entity).created_by_service
+            if entity.entity_id.startswith("group.") and not entity.created_by_service
```

Comment:
Aren't `Group` entities required to have an entity_id that starts with `"group."`? Why do we need that check?

Reply:
Need to double check that platform entities don't end up in that bucket as well but I think you are right that the check isn't likely redundant

Reply:
Platform group entities don't belong to this EntityComponent. They belong to fan, switch etc entity components.

Reply:
In that case it's safe to drop the redundant check